### DISCUSSION TITLE
[v5] AsyncHttpContext

### DISCUSSION THREAD

**Discussion by edesideri** [adonisjs/core#1913] [General]
Hi,
i read this https://github.com/adonisjs/lucid/pull/555 and i would use AsyncHttpContext in my model hook like this https://github.com/adonisjs/lucid/pull/555#issuecomment-643589356
i  add Adonis/Core/AsyncHttpContextMiddleware to my global middleware in start/kernel.ts as suggested from @mastermunj and now i get this error
`IocLookupException: E_IOC_LOOKUP_FAILED: Cannot resolve "Adonis/Core/AsyncHttpContextMiddleware" namespace from the IoC Container

at C:\webapp\htdocs\adonis\server\start\kernel.ts(anonymous):23
18  |
19  | An array of global middleware, that will be executed in the order they
20  | are defined for every HTTP requests.
21  |
22  */
23  Server.middleware.register([
24      'Adonis/Core/BodyParserMiddleware',
25      'Adonis/Core/AsyncHttpContextMiddleware',
26  ])
27
28  /*

1 Function.lookupFailed
  C:\webapp\htdocs\adonis\server\node_modules\@adonisjs\fold\build\src\Exceptions\IocLookupException.js:18

2 Ioc.onLookupFailed
  C:\webapp\htdocs\adonis\server\node_modules\@adonisjs\fold\build\src\Ioc\index.js:61

3 IocResolver.resolve
  C:\webapp\htdocs\adonis\server\node_modules\@adonisjs\fold\build\src\Resolver\index.js:89

4 MiddlewareStore.resolveMiddleware
  C:\webapp\htdocs\adonis\server\node_modules\@adonisjs\http-server\build\src\MiddlewareStore\index.js:61

5 MiddlewareStore.register
  C:\webapp\htdocs\adonis\server\node_modules\@adonisjs\http-server\build\src\MiddlewareStore\index.js:68

6 Module._compile
  C:\webapp\htdocs\adonis\server\node_modules\pirates\lib\index.js:99

[ warn ]  Underlying HTTP server died with "0 code"`

can you help me?
thanks

**Comment by RomainLanz** [comment]
Hey @edesideri! 👋 

The `AsyncHttpContext` is a highly experimental feature that is not added to the core at the moment.
This is why you cannot resolve it from the IoC.

For the moment, if you want to use is in your application, you can install [the following package](https://github.com/RomainLanz/async-http-context).

Also, you can follow [this PR](https://github.com/adonisjs/http-server/pull/18) to know when (or if) it's going to be added to the core.

Take notes that Targos has a different build to make it work in Model's Hook.

**Reply by edesideri** [reply]
Ok its all clear now!
thanks a lot
